pkg/core/fetch.go

```diff
@@ -311,8 +311,9 @@ func runProviderFetch(ctx context.Context, pm *plugin.Manager, info ProviderInfo
 	}
 
 	pLog.Info().Msg("provider configured successfully")
-	summary, diags := executeFetch(ctx, pLog, providerPlugin, info, metadata, opts.UpdateCallback)
-	return summary, convertToFetchDiags(diags, info.Provider.Name, providerPlugin.Version())
+	summary, fetchDiags := executeFetch(ctx, pLog, providerPlugin, info, metadata, opts.UpdateCallback)
+	fetchDiags.Add(convertToConfigureDiags(resp.Diagnostics))
+	return summary, convertToFetchDiags(fetchDiags, info.Provider.Name, providerPlugin.Version())
```

Comment:
on second thought I think the best bet is to convert them to fetch/configure diags separately and do a final `return summary, diags1.Add(diags2)`